src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationMigrationIntTest.kt

```diff
@@ -407,4 +410,250 @@ class AllocationMigrationIntTest : SqsIntegrationTestBase() {
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `must have correct role to cancel a migration`() {
+      webTestClient.post().uri("/migrate/allocations/{migrationId}/cancel", "some id")
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_BANANAS")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `will return a not found if no running migration found`() {
+      webTestClient.post().uri("/migrate/allocations/{migrationId}/cancel", "some id")
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `will cancel a running migration`() {
+      stubMigrationDependencies(entities = 10)
+      mappingApi.stubAllocationsMappingByMigrationId(count = 10)
+
+      val migrationId = webTestClient.post().uri("/migrate/allocations")
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
+        .header("Content-Type", "application/json")
+        .body(
+          BodyInserters.fromValue(
+            """
+            {
+              "prisonId": "MDI"
+            }
+            """.trimIndent(),
+          ),
+        )
+        .exchange()
+        .expectStatus().isAccepted
+        .returnResult<MigrationContext<ActivitiesMigrationFilter>>()
+        .responseBody.blockFirst()!!.migrationId
+
+      webTestClient.post().uri("/migrate/allocations/{migrationId}/cancel", migrationId)
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isAccepted
+
+      webTestClient.get().uri("/migrate/allocations/history/{migrationId}", migrationId)
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.migrationId").isEqualTo(migrationId)
+        .jsonPath("$.status").isEqualTo("CANCELLED_REQUESTED")
+
+      await atMost Duration.ofSeconds(60) untilAsserted {
+        webTestClient.get().uri("/migrate/allocations/history/{migrationId}", migrationId)
+          .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
+          .header("Content-Type", "application/json")
+          .exchange()
+          .expectStatus().isOk
+          .expectBody()
+          .jsonPath("$.migrationId").isEqualTo(migrationId)
+          .jsonPath("$.status").isEqualTo("CANCELLED")
+      }
+    }
+  }
+
+  @Nested
+  @DisplayName("GET /migrate/allocations/active-migration")
+  inner class GetActiveMigration {
+    @BeforeEach
+    internal fun createHistoryRecords() {
+      runBlocking {
```

Comment:
Peter suggested that we should now be using runTest...quote from an earlier review of mine:
think we're trying to use `runTest` instead of `runBlocking` as it skips any delays (but not sure there will be any difference here)

Reply:
I don't really understand the benefit of runTest but it does seem like current best practise - will do in next PR.